### ISSUE TITLE
R API on the server

### ISSUE THREAD

**Issue by AnastasiaDv491** [osoc21/technical-support#3]
**Is your request related to a problem? Please describe.**
connecting to a server like Netlify (since our team doesnt have an experience  with it)

**Describe the solution you'd like**
Ability to push R based API to the server.

**Describe alternatives you've considered**
Havent considered much becuase we are a bit clueless on what to do for the server

**Comment by jbelien** [comment]
I don't think it's possible to deploy R code on Netlify.

I would suggest to use Docker (and a [Dockerfile](https://docs.docker.com/engine/reference/builder/)).
Once your Dockerfile is ready (in [your repository](https://github.com/osoc21/Future-of-Work)), ping me here and I'll setup an automated deployment for it using [Digital Ocean App Platform](https://www.digitalocean.com/products/app-platform/).

**Comment by AnastasiaDv491** [comment]
thanks ! will try to figure it out

**Comment by ErinnVdSande** [comment]
@jbelien I have created a docker-compose and a dockerfile but I have never worked with the technology before so I can only get it running with the compose right now. Is that good enough?

**Comment by jbelien** [comment]
> @jbelien I have created a docker-compose and a dockerfile but I have never worked with the technology before so I can only get it running with the compose right now. Is that good enough?

I won't need the `docker-compose.yml` file. I just need the `Dockerfile`.
Locally you can use [`docker compose`](https://docs.docker.com/engine/reference/commandline/compose/) if you want.
You can also use [`docker build`](https://docs.docker.com/engine/reference/commandline/build/) + [`docker run`](https://docs.docker.com/engine/reference/commandline/run/) ; that's what I will use.

Is this already pushed to somewhere I can check ?

**Comment by ErinnVdSande** [comment]
It's on the https://github.com/osoc21/Future-of-Work on the hackaton branch in the backend folder

**Comment by jbelien** [comment]
Your [`dockerfile`](https://github.com/osoc21/Future-of-Work/blob/3babdc6521c4a16b72df13c80d09c16539d8726f/docker-compose.yml) seems correct (even though I would rename it `Dockerfile`).

As I said, I don't need your `docker-compose.yml` file.
Could you try the following (I didn't test it, might need some tweaking) ?

```cmd
docker build ./backend/ --no-cache -t api-service
docker run --publish 7000:8000 --volumes ./backend/api.R:/app/api.R api-service
```

The first command builds an image called "api-service".
The second command runs the image with the parameters you chose.

**Comment by ErinnVdSande** [comment]
I get an error on the second line 

`docker run --publish 7000:8000 --volumes ./backend/api.R:/app/api.R api-service`

unknown flag: --volumes
See 'docker run --help'.

**Comment by jbelien** [comment]
Indeed, it's `--volume` instead of `--volumes`.

```cmd
docker run --publish 7000:8000 --volume ./backend/api.R:/app/api.R api-service
```

Please read [`docker build`](https://docs.docker.com/engine/reference/commandline/build/) and [`docker run`](https://docs.docker.com/engine/reference/commandline/run/) documentation.

**Comment by jbelien** [comment]
Side note, I think that `COPY` (see [documentation](https://docs.docker.com/engine/reference/builder/#copy)) in your `Dockerfile` is the way to go (instead of `--volume`).

**Comment by ErinnVdSande** [comment]
Yeah I changed it to a `COPY` and pushed it to the branch the commands i use now are:
`docker build ./backend/ --no-cache -t api-service`
and
`docker run --publish 7000:8000 api-service`

**Comment by archived-m** [comment]
@jbelien I think this can be closed given that we moved away from R in its entirety. We'll open a new issue when we're ready to deploy and our Dockerfile is up to snuff :)
